test/cli-e2e-specs.js

```diff
@@ -142,4 +143,87 @@ describe('Driver CLI', function () {
       await fs.exists(installPath).should.eventually.be.false;
     });
   });
+
+  describe('run', function () {
+    it('should run a valid driver, valid script, and result in success', async function () {
+      await clear();
+      const driverName = 'fake';
+      const scriptName = 'fake-success';
+      await run('install', [localFakeDriverPath, '--source', 'local', '--json']);
+      const out = JSON.parse(await run('run', [driverName, scriptName, '--json']));
+      out.should.not.include.key('error');
+    });
+    it('should run a valid driver, valid error prone script, and return error in json', async function () {
+      await clear();
+      const driverName = 'fake';
+      await run('install', [localFakeDriverPath, '--source', 'local', '--json']);
+      const out = JSON.parse(await run('run', [driverName, 'fake-error', '--json']));
+      out.should.include.key('error');
+    });
+    it('should take a valid driver, invalid script, and throw an error', async function () {
+      await clear();
+      const driverName = 'fake';
+      await run('install', [localFakeDriverPath, '--source', 'local', '--json']);
+      await chai.expect(run('run', [driverName, 'foo', '--json'])).to.eventually.be.rejectedWith(Error);
+    });
+    it('should take an invalid driver, invalid script, and throw an error', async function () {
+      await clear();
+      const driverName = 'foo';
+      await chai.expect(run('run', [driverName, 'bar', '--json'])).to.eventually.be.rejectedWith(Error);
+    });
+  });
 });
+
+describe('Plugin CLI', function () {
+  let appiumHome;
+
+  before(async function () {
+    appiumHome = await tempDir.openDir();
+  });
+
+  after(async function () {
+    await fs.rimraf(appiumHome);
+  });
+
+  async function clear () {
+    await fs.rimraf(appiumHome);
+    await mkdirp(appiumHome);
+  }
+
+  async function run (pluginCmd, args = [], raw = false) {
+    args = [...args, '--appium-home', appiumHome];
+    const ret = await exec('node', ['.', 'plugin', pluginCmd, ...args], {cwd});
+    if (raw) {
+      return ret;
+    }
+    return ret.stdout;
+  }
+
+  describe('run', function () {
+    it('should run a valid plugin, valid script, and result in success', async function () {
+      await clear();
+      const pluginName = 'fake';
+      const scriptName = 'fake-success';
+      await run('install', [fakePluginDir, '--source', 'local', '--json']);
+      const out = JSON.parse(await run('run', [pluginName, scriptName, '--json']));
+      out.should.not.include.key('error');
+    });
+    it('should run a valid plugin, valid error prone script, and return error in json', async function () {
+      await clear();
```

Comment:
in this and the next test, we theoretically don't need to clear and install, right? what about just clearing and installing once in a `before` for this whole describe?